docs(react-tabster): adds stories for focus indicator documentation (#23239)
microsoft/fluentui: packages/react-components/react-components/src/Concepts/Accessibility/FocusIndicator.stories.mdx

## Focus indicator

Fluent UI components use [tabster](https://github.com/microsoft/tabster) for focus handling functionality, so that they can be easily integrated with application-level tabster functionality such as delooser and cross-iframe focusing.

Fluent UI `@fluentui/react-tabster` package defines `useKeyboardNavAttribute`, `createCustomFocusIndicatorStyle` and `createFocusOutlineStyle` to integrate tabster keyboard navigation mechanism seemlesly within Fluent UI.

### useKeyboardNavAttribute

Instantiates [keyborg](https://github.com/microsoft/keyborg) and add attribute to ensure focus indicator synced to keyborg logic.

### Styling focus indicators

Both methods are used internally to declare focus indication based on the attribute provided by `useKeyboardNavAttribute`, by default a component on Fluent UI uses `createFocusOutlineStyle` to create a focus indicator with [outline style](https://developer.mozilla.org/en-US/docs/Web/CSS/outline), but in some cases an outline is not the desired style, on those cases `createCustomFocusIndicatorStyle` can be used to add a cusom style with the same mechanisms.

#### createFocusOutlineStyle

> For example, [AccordionHeader](?path=/docs/components-accordion--default) component is using `createFocusOutlineStyle` to a the default outline style when focus is detected

Reviewer: ```suggestion
For example, the [AccordionHeader](?path=/docs/components-accordion--default) component uses `createFocusOutlineStyle` to style the default outline style when focus is detected
```

<Canvas>
  <Story id="components-accordion--default" />
</Canvas>

```tsx

> import { makeStyles } from '@griffel/react';

Reviewer: ```suggestion
import { makeStyles } from '@fluentui/react-components';
```

you might need to double check but I'm pretty sure Griffel is reexported

import { createFocusOutlineStyle } from '@fluentui/react-tabster';

> const useStyles = makeStyles({

Reviewer: please also add some code where this classsname is applied

>   focusIndicator: createFocusOutlineStyle(),

Reviewer: I would also add some docs and examples to mention that this helper is somewhat customizable

```

> #### createCustomFocusIndicatorStyle

Reviewer: ```suggestion
#### createCustomFocusIndicatorStyle

> ⚠️ A bad focus indicator can have serious accessibility consequences and can render
> your experience unusable by certain user. Please ensure before creating a custom focus
> indicator that you have gotten the necessary feedback from designers and accessibility
> experts.
```

For example, [Link](?path=/docs/components-link--default) component is using `createCustomFocusIndicatorStyle` to add a double underlined focus indication style

<Canvas>
  <Story id="components-link--default" />
</Canvas>

```tsx
import { makeStyles } from '@griffel/react';

> import { createCustomFocusIndicatorStyle } from '@fluentui/react-tabster';

Reviewer: ```suggestion
import { createCustomFocusIndicatorStyle } from '@fluentui/react-components';
```

let's use the recommended export for our customers

import { Meta, Story, Canvas } from '@storybook/addon-docs';
import { ScenariosListLink } from '../../AccessibilityScenarios/utils';

<Meta title="Concepts/Developer/Accessibility/Focus indicator" />

## Focus indicator

Fluent UI components use [tabster](https://github.com/microsoft/tabster) for focus handling functionality, so that they can be easily integrated with application-level tabster functionality such as delooser and cross-iframe focusing.

Fluent UI `@fluentui/react-tabster` package defines `useKeyboardNavAttribute`, `createCustomFocusIndicatorStyle` and `createFocusOutlineStyle` to integrate tabster keyboard navigation mechanism seemlesly within Fluent UI.

### useKeyboardNavAttribute

Instantiates [keyborg](https://github.com/microsoft/keyborg) and add attribute to ensure focus indicator synced to keyborg logic.

> `useKeyboardNavAttribute` is being used internally by Fluent UI providers and portals to ensure that top level elements will have attributes to indicate focus, in a similar way to what `:focus-visible` does, ensuring Fluent UI components are aware when focus is required.

Reviewer: ```suggestion
```

I'm sorry but I cannot understand this paragraph/sentence

> ### createCustomFocusIndicatorStyle and createFocusOutlineStyle

Reviewer: ```suggestion
### Styling focus indicators
```

You have subheadings later for each of these functions, let's be more mission focused in the title

> Both methods are used internally to declare focus indication based on the attribute provided by `useKeyboardNavAttribute`, by default a component on Fluent UI uses `createFocusOutlineStyle` to create a focus indicator with [outline style](https://developer.mozilla.org/en-US/docs/Web/CSS/outline), but in some cases an outline is not the desired style, on those cases `createCustomFocusIndicatorStyle` can be used to add a cusom style with the same mechanisms.

Reviewer: ```suggestion
The default focus indicator used in Fluent UI is an outline. However, in some cases more specific focus indicators are necessary depending on the use case and component design. In order to accommodate these requirements, Fluent UI exports two different utilities to style focus indicators:

1. `createFocusOutlineStyle`
2. `createCustomFocusIndicatorStyle`

Both of the helper functions are powered using the method described above.
```
